<launch>
        
    <param name="robot_description" command="$(find xacro)/xacro $(find franka_description)/robots/panda_arm.urdf.xacro hand:=true" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" output="screen">        
        <rosparam param="source_list"> [franka_rpm/joint_states] </rosparam>
        <param name="rate" value="30"/>
    </node>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pose_optimization)/launch/ycbv.rviz" />
    
    <rosparam command="load" file="$(find pose_optimization)/config/ycbv_object.yaml" />
    <rosparam command="load" file="$(find pose_optimization)/config/default.yaml" />     
    <node pkg="pose_optimization" type="graph_optimization" name="graph_optimization" output="screen" />    
    <node pkg="pose_optimization" type="visualization" name="visualization" output="screen" />
                
</launch>
